emulator.c

```diff
@@ -302,6 +326,16 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
       }
     case 0xc6: // NOLINT
       printf("ADI ");
+      // Affects Z, S, P, CY, AC
+      uint8_t immediate = cpu_read_mem(cpu, cpu->pc + 1);
+      uint16_t answer = cpu->a + immediate;
+      update_zero_flag(cpu, (uint8_t)answer);
+      update_sign_flag(cpu, (uint8_t)answer);
+      update_parity_flag(cpu, (uint8_t)answer);
+      update_carry_flag(cpu, answer > 0xFF); // NOLINT
+      update_aux_carry_flag(cpu, cpu->a, immediate);
+      cpu->a = (uint8_t)answer;
+      cpu->pc += 2;
```

Comment:
Think you only need to increment pc by 1 here as one is captured by default at the end of the switch statement.